### PR TITLE
[WIP] some more performance optimizations

### PR DESCRIPTION
**Pull request by alexcjohnson**
Following on #1772 - some more poking around for performance bottlenecks.

### CODE REVIEW

File: src/lib/svg_text_utils.js [hunk 1]
```diff
@@ -106,9 +65,7 @@ exports.convertToTspans = function(_context, gd, _callback) {
                 'white-space': 'pre'
             });
 
-        result = _context.appendSVG(converted);
-
-        if(!result) _context.text(str);
+        _context.node().innerHTML = convertToSVG(str);
```
Comment (etpinard) [inline src/lib/svg_text_utils.js]: Hmm. Does this work in IE?

Comment (etpinard) [inline src/lib/svg_text_utils.js]: http://caniuse.com/#search=innerHTML

image

Inconclusive. We'll have to try it on Browserstack.

**Comment by etpinard** [comment]
If this works out, this PR is :trophy: 

This should get us pretty close to plotly.js working in jsdom out of the box, referencing:

- https://gist.github.com/etpinard/bee7d62b43b6bb286950
- https://github.com/plotly/plotly.js/issues/1675

File: src/lib/svg_text_utils.js (hunk 1 above)
Comment (alexcjohnson) [inline src/lib/svg_text_utils.js]: oh crap you're right... fails even in IE11

**Comment by alexcjohnson** [comment]
Alright, I suppose I could write some sort of feature detection to enable or disable this behavior... but I think the better solution (which with any luck will be even faster as well as cross-browser!) will be to just build elements like we do for everything else, rather than building a text string and turning that into elements. Going to give that a shot.

**Pull request closed without merging**